openmicroscopy/openmicroscopy
# Table download

This PR has now been ported to https://github.com/ome/omero-web/pull/3 but tests remain in this one.

In several training workshops, explaining what an OMERO.table is and what data is contains is made difficult because it is very hard to show a table. Even when downloaded, it is not easy see the contents of a table since it's an HDF file.
For users of tables, inspecting the data they contain for trouble-shooting purposes etc is also hard. It is only possible to see the data for a single object/row at a time.

This allows users to view an OMERO.table file as an html table and to download it as a CSV.

# Testing this PR

1. Find an OMERO.table attached to an object e.g. Project, Screen etc. Tables are recognised by the ```openmicroscopy.org/omero/bulk_annotations``` namespace (you can find objects annotated with this namespace by searching for 'bulk'. E.g. http://web-dev-merge.openmicroscopy.org/webclient/search/?search_query=bulk (user-3)
2. Files will show an 'eye' (view) icon. When clicked, this opens the file as an html table in another tab.
3. The table page also has links to download the table as CSV (or show as JSON)
4. Click to download CSV and open in Excell.
5. Also click to view JSON (might be useful for devs?)
6. Check new web test is passing

NB: the default action of clicking the OMERO.table annotation itself to download the HDF file remains the same. Do we want the default action to open the html table (with links there allowing download as HDF or CSV) since download as HDF is rarely very useful?
